include/boost/geometry/algorithms/detail/disjoint/point_geometry.hpp

```diff
@@ -39,20 +39,25 @@ namespace detail { namespace disjoint
 {
 
 
-template<typename Point, typename Geometry>
-struct disjoint_point_linear
+struct disjoint_point_segment
 {
-    static inline
-    bool apply(Point const& pt, Geometry const& g)
+    template <typename Point, typename Segment>
+    static inline bool apply(Point const& point, Segment const& segment)
     {
-        return !geometry::covered_by(pt, g);
+        typedef geometry::model::referring_segment<Point const> other_segment;
+
+        other_segment other(point, point);
+        return disjoint_segment
+            <
+                Segment, other_segment
+            >::apply(segment, other);
```

Comment:
What I had in mind is, that `reverse_covered_by` could also be used for Pt/Seg. AFAIU there is no need to have `disjoint_point_segment` becasue assuming that `covered_by()` works for Pt/Seg, `reverse_covered_by` should work out of the box. Or am I missing something?